### PR TITLE
Di ingectable

### PR DESCRIPTION
**Pull request by dmitry-rachko-idf**

### CODE REVIEW

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: lib/di/injector.config.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
```
Comment (sergei-mikhailovskii-idf) [inline lib/di/injector.config.dart:1]: gitignore

File: packages/presentation/lib/bloc/home_bloc.dart [hunk 2]
```diff
@@ -30,7 +34,7 @@ class _HomeBloc extends BlocImpl implements HomeBloc {
     updateData();
 
     _screenData.isPalindrome =
-        await _palindromeUseCase.isPalindrome(_screenData.stringInput);
+        await _palindromeUseCase.call(_screenData.stringInput);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/bloc/home_bloc.dart:37]: fyi: it's not obvious to call `call` method explicitly. u can call _palindromeUseCase(_screenData.stringInput)

File: packages/presentation/pubspec.lock [hunk 3]
```diff
@@ -1,6 +1,27 @@
 # Generated by pub
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/pubspec.lock]: ignore

File: packages/presentation/pubspec.yaml [hunk 4]
```diff
@@ -14,6 +14,11 @@ dependencies:
     sdk: flutter
   domain:
     path: ../domain
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/pubspec.yaml]: remove domain, as data has it's own link to domain

File: pubspec.lock [hunk 5]
```diff
@@ -1,27 +1,97 @@
 # Generated by pub
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.lock]: ignore

File: packages/presentation/pubspec.yaml [hunk 6]
```diff
@@ -14,6 +14,11 @@ dependencies:
...
+  data:
+    path: ../data
+  
+  get_it: ^7.2.0
+  injectable: ^1.5.3
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/pubspec.yaml]: no need in specifying dependencies in each module. u can specify it only in one and other modules, which include the module with dependency will also have this dependency. so, if u'll add get_it to the domain, u'll have access to it in data, domain and presentation. read about transitive dependencies in the flutter

File: pubspec.yaml [hunk 7]
```diff
@@ -31,12 +31,19 @@ dependencies:
     sdk: flutter
   presentation:
     path: packages/presentation
+  domain:
+    path: packages/domain
+  data:
+    path: packages/data
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.yaml]: u can specify only presentation

File: pubspec.yaml [hunk 8]
```diff
@@ -31,12 +31,19 @@ dependencies:
...
 
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
  
+  get_it: ^7.2.0
+  injectable: ^1.5.3
```
Comment (sergei-mikhailovskii-idf) [inline pubspec.yaml]: remove it

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**